Documentation/2.0/api.md

```diff
@@ -286,6 +287,13 @@ curl 'http://127.0.0.1:2379/v2/keys/foo?wait=true&waitIndex=7'
 
 The watch command returns immediately with the same response as previously.
 
+**Note**: etcd only keeps the responses of the most recenet 1000 events. 
+It is recommended to send the response to another thread to process immediately
+instead of blocking the watch while processing the result. 
+If we missed all the 1000 events, we need to recover the current state of the 
+watching key space. Typically, we do a get and then watch again from the (etcdIndex + 1)
```

Comment:
Instead of "typically", I say we strongly tell them what do. "First, fetch the current state with a get and start the watch at etcdIndex + 1."